lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -696,6 +649,14 @@ TypeLoweringVisitor::addArg(Operation *module, unsigned insertPt,
       filterAnnotations(context, oldArg.annotations.getArrayAttr(), srcType,
                         field, needsSym, sym);
 
+  if (field.type.isGround() && oldArg.sym && !oldArg.sym.getValue().empty()) {
+    auto moduleLike = cast<FModuleLike>(module);
+    auto key = hw::InnerRefAttr::get(moduleLike.moduleNameAttr(), oldArg.sym);
+    assert(insertPt >= insertPtOffset + 1 && "insertPtOffset is too large");
+    auto value = PortAnnoTarget(module, insertPt - 1 - insertPtOffset);
+    innerRefRenames[key].push_back(value);
+  }
```

Comment:
The mechanics of how the port lowering works is that a module port list consisting of all old and new ports are created. After this, old ports are removed. This necessitates getting the final port index _after dead ports are deleted_ into `lowerArg` and `addArg` in order to populate `innerRefRenames` for ports. I would have preferred to not pass in another numeric argument, but the refactoring to get the code to work by maintaining separate old and new port lists proved tedious.